src/test/ui/stability-attribute/allow-through-unstable-misuse.rs

```diff
@@ -0,0 +1,8 @@
+#![crate_type = "lib"]
+#![feature(staged_api)]
+#![feature(rustc_attrs)]
+#![stable(feature = "foo", since = "1.0.0")]
+
+#[unstable(feature = "bar", issue = "none")]
+#[rustc_allowed_through_unstable_modules]
+pub struct UnstableType(());
```

Comment:
I based my attribute loosely on `#[rustc_promotable]` and noticed in the process that it is exempted from the requirement to add ui tests for every error code. Rather than just cargo cult that I decided to try testing mine but I'm running into an issue where it emits the diagnostic twice...

Reply:
actually reversed that decision for now so I can focus on the rest of the PR, will revisit this once I'm happy with everything else.